### PR TITLE
Restores old coin functionality as an emag thing

### PR DESCRIPTION
**Pull request by Iamgoofball**
at the request of Emily

### CODE REVIEW

**Comment by UrsulaMejor** [comment]
Putting the old functionality back in as an emag easter egg is an good compromise

**Comment by ZeWaka** [comment]
I'd much rather it gain the 'old' bugged behaviour after being emagged, rather than the other way around. Almost like you're loading/magnetizing the coin.

File: code/obj/item/coin.dm [hunk 1]
```diff
@@ -19,14 +20,30 @@
 	..(hit_atom)
 	flip()
 		
+		
+/obj/item/coin/emag_actemag_act(var/mob/user, var/obj/item/card/emag/E)
```
Comment (zjdtmkhzt) [inline code/obj/item/coin.dm:24]: emag_actegmag_act

Comment (zjdtmkhzt) [inline code/obj/item/coin.dm:24]:
```suggestion
/obj/item/coin/emag_act(var/mob/user, var/obj/item/card/emag/E)
```

File: code/obj/item/coin.dm [hunk 2]
```diff
@@ -19,14 +20,30 @@
...
+		boutput(user, "You magnetize the coin, ruining it's chances of ever being used in the Inter-galactic Poker Tournaments ever again.")
+		emagged = TRUE
+		
 /obj/item/coin/proc/flip()
-	if(prob(1))
-		src.visible_message("<span class='notice'>The coin lands on its side. Fuck.</span>")
+	if(!emagged)
+		if(prob(1))
+			src.visible_message("<span class='notice'>The coin lands on its side. Fuck.</span>")
+			return
```
Comment (zjdtmkhzt) [inline code/obj/item/coin.dm:34]: I think this return may be unneeded?

Comment (zjdtmkhzt) [inline code/obj/item/coin.dm:34]:
```suggestion
```

**Pull request merged**